### PR TITLE
Move Layout browser into Top Nav

### PR DESCRIPTION
**Pull request by 2metres**
**User-Facing Changes**
None

**Description**
Move Layout browser into Top Nav behind new UI feature flag

Screen Shot 2023-03-02 at 2 32 19 pm

### CODE REVIEW

**Comment by 2metres** [comment]
Need work:

Clicking sign up banner/alert no longer works

**Comment by foxymiles** [comment]
Looks good. I think selecting items from the menus should close those menus.

File: packages/studio-base/src/components/LayoutBrowser/SignInPrompt.tsx [hunk 1]
```diff
@@ -3,60 +3,60 @@
...
-            <CloseIcon />
-          </IconButton>
-        )
-      }
-    />
+    <ButtonBase className={classes.root} onClick={openAccountSettings}>
+      <Typography align="left" className={classes.title} variant="body2">
+        <Link color="inherit" onClick={openAccountSettings} underline="always">
+          Sign in
+        </Link>{" "}
```
Comment (jtbandes) [inline packages/studio-base/src/components/LayoutBrowser/SignInPrompt.tsx:44]: Does this button actually work? It didn't seem to work for me on desktop.

**Comment by amacneil** [comment]
product review feedback
- keep layout icon, can potentially test showing layout name in future PR
- prefer the dropdown scrolling on main vs this branch
- tighter menu/list item spacing across the whole app (separate PR)
- settings could go under avatar menu (separate PR)

File: packages/studio-base/src/components/AppBar/AddPanel.tsx [hunk 2]
```diff
@@ -3,17 +3,26 @@
...
 import { Menu } from "@mui/material";
+import { makeStyles } from "tss-react/mui";
 
 import PanelList from "@foxglove/studio-base/components/PanelList";
 import useAddPanel from "@foxglove/studio-base/hooks/useAddPanel";
 
+const useStyles = makeStyles()({
+  paper: {
+    // default spacing + top nav height + playback bar
+    maxHeight: "calc(100% - 32px - 48px - 48px)",
```
[the remark this answers is not in the document]
Comment (2metres) [reply packages/studio-base/src/components/AppBar/AddPanel.tsx:14]: Apparently from Product Review we don't

**Comment by 2metres** [comment]
Weird problem flash when switching layouts

https://user-images.githubusercontent.com/924528/222297014-13a000a5-1b3b-47dc-aff5-64d97e5f8e7c.mov

**Comment by defunctzombie** [comment]
I would remove "from JSON" in the import option - you can put "from file" if you want. I'd also consider moving this to the end of the list? Its not the common flow is it?

File: packages/studio-base/src/components/AppBar/Help.tsx [hunk 3]
```diff
@@ -30,35 +22,40 @@ const useStyles = makeStyles()((theme) => ({
...
+  open,
+  transformOrigin,
+}: HelpMenuProps): JSX.Element {
   const { classes } = useStyles();
   const currentUserType = useCurrentUserType();
   const analytics = useAnalytics();
 
   return (
     <Menu
-      {...menuProps}
```
Comment (foxymiles) [inline packages/studio-base/src/components/AppBar/Help.tsx:61]: ❤️

**Review by foxymiles — APPROVED** [review]

File: packages/studio-base/src/components/LayoutBrowser/SignInPrompt.tsx (hunk 1 above)
Comment (2metres) [reply packages/studio-base/src/components/LayoutBrowser/SignInPrompt.tsx]: Merging in @foxymiles's PR to fix

**Pull request merged**